profiler/internal/immutable/stringslice.go

```diff
@@ -0,0 +1,33 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package immutable provides read-only types
+package immutable
+
+// StringSlice is a wrapped slice which cannot be modified and must be copied to
+// access. The zero value for a StringSlice is ready to use. All StringSlice
+// methods are safe to call from multiple goroutines concurrently.
+type StringSlice struct {
+	s []string
+}
+
+// NewStringSlice wraps a copy of the input slice
```

Comment:
nick: maybe remove reference to wrapping, because it's not the original slice that gets "wrapped", a copy is being made.

Reply:
Done